### ISSUE TITLE
Tasks

### ISSUE THREAD

**Issue by candroid-man** [streetwriters/notesnook#171]
**Request**

Combine notes with a to-do app, and you will be unstoppable! The checklist within the app already are not good enough, you could have the preview be a progress bar like in Standard Notes, and be able to have due dates, reminders, sub-tasks, etc. Basically, make Todoist but better for privacy. You could make it where you can make a to-do list anywhere (in a notebook, topic, etc.), but I think it would be better for this to have its own section of the app, and then you could potentially add a to-do list to a note.

**Additional context**

This would be a HUGE selling point for the average user who isn't concerned about privacy and is not very technically handy.

**Comment by Tamnac** [comment]
Agreed, I currently have not entirely switched to notesnook, because it comes nowhere close to [Amplenote](https://www.amplenote.com) in terms of task management. You don't need to go that far, but having due dates, reminders, subtasks, maybe a master list of tasks etc. would mean a lot.

**Comment by candroid-man** [comment]
Just want to leave this here so I don't forget: tasks should be completely separate from notes, but having them in the same app would be best I think. But it would be cool if you could attach notes to tasks somehow

**Comment by thecodrr** [comment]
We have already taken steps towards this in the new editor. Tasks will be part of notes as notes are the fundamental building blocks in Notesnook. They will eventually include things like reminders, due dates etc.

**Comment by candroid-man** [comment]
I still think you should consider #412 tasks menu. This way you wouldn't be absolutely forced to have a dedicated note taking up space strictly for tasks. You would need to add a toggle to each checklist in every note whether or not to display that list in the task menu or something similar, but I really think it would be a good blend between a notes and tasks app
